presto-docs/src/main/sphinx/optimizer/reorder-joins.rst

```diff
@@ -0,0 +1,30 @@
+*************
+Reorder joins
+*************
+
+Cross joins elimination
+-----------------------
+
+
+When config property ``reorder-joins`` (or ``reorder_joins`` session property) is enabled optimizer will search for cross joins
+in query plan and try to eliminate them by changing order of the joins. While doing so optimizer will try to preserve
+original join order as much as possible.
+
+For example for the following query:
+
+.. code-block:: sql
+
+    SELECT * FROM part p, orders o, lineitem l WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey;
+
+Original join order ``part, orders, lineitem`` means that presto first have to join ``part`` table with ``orders`` table,
+for which there is no join condition. ``reorder-joins`` enabled changes join order to ``part, lineitem, orders``
+which eliminates this cross join.
+
+For following query:
+
+.. code-block:: sql
+
+    SELECT * FROM part p, orders o, lineitem l, supplier s, nation n
+    WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey AND l.suppkey = s.suppkey AND s.nationkey = n.nationkey;
+
+Join order will change from ``part, orders, lineitem, supplier, nation`` to ``part, lineitem, orders, supplier, nation``.
```

Comment:
Please add a "Note: the same can be achieved by manually changing the order of tables in the query". Just in case it's not obvious for some users :)